pkg/login/social/generic_oauth.go

```diff
@@ -412,33 +434,75 @@ func (s *SocialGenericOAuth) FetchPrivateEmail(client *http.Client) (string, err
 	return email, nil
 }
 
-func (s *SocialGenericOAuth) FetchTeamMemberships(client *http.Client) ([]int, bool) {
+// FetchTeamMemberships will attempt to get the user's team string IDs from the JWT claims using the 'team_id_attribute_path' setting.
+// If the attribute path does not find anything in the JWT claims, then the same attribute path is used to attempt to retrieve the team IDs from the OIDC '/userinfo' endpoint.
+// If the attribute path still does not find anything in the '/userinfo' response, then the same attribute path is used to attempt to retrieve the team IDs from the 'teams_url'
+// If team IDs are found, they are returned. Otherwise, an error is returned.
+func (s *SocialGenericOAuth) FetchTeamMemberships(client *http.Client) ([]string, error) {
+	var err error
+	var ids []string
+
+	if s.teamsUrl == "" {
+		ids, err = s.fetchTeamMembershipsFromDeprecatedTeamsUrl(client)
+	} else {
+		ids, err = s.fetchTeamMembershipsFromTeamsUrl(client)
+	}
+
+	if err == nil {
+		s.log.Debug("Received team memberships", "ids", ids)
+	}
+
+	return ids, err
+}
+
+func (s *SocialGenericOAuth) fetchTeamMembershipsFromDeprecatedTeamsUrl(client *http.Client) ([]string, error) {
+	s.log.Warn("[Deprecated] Switch to the use of the teams_url and team_id_attribute_path configuration options for filtering on team membership")
```

Comment:
I don't think we should treat this as deprecated because basically every configured Generic OAuth provider would get this notice logged. I think we should still provide the same experience as before to provide a good backward compatibility story when teams_url is not configured and basically fallback to the old behavior.